util/reggen/ip_block.py

```diff
@@ -181,21 +166,30 @@ def __init__(self,
                  inter_signals: List[InterSignal],
                  bus_interfaces: BusInterfaces,
                  clocking: Clocking,
-                 xputs: Tuple[Sequence[Signal],
-                              Sequence[Signal],
+                 xputs: Tuple[Sequence[Signal], Sequence[Signal],
                               Sequence[Signal]],
                  wakeups: Sequence[Signal],
                  reset_requests: Sequence[Signal],
                  expose_reg_if: bool,
                  scan_reset: bool,
                  scan_en: bool,
-                 countermeasures: List[CounterMeasure]):
+                 countermeasures: List[CounterMeasure],
+                 node: str = ''):
         assert reg_blocks
 
+        # Filter the interfaces and reg_blocks if request to build only for a
+        # specific reg_block node.
+        dev_if_names = []  # type: List[Optional[str]]
+        if node:
+            dev_if_names += [
+                i for i in bus_interfaces.named_devices if i == node
+            ]
+            reg_blocks = {k: v for k, v in reg_blocks.items() if k == node}
+        else:
+            dev_if_names += bus_interfaces.named_devices
```

Comment:
We probably ought to change this to filter on outputs, rather than make an `IpBlock` not represent the entire block, hehe

Future cleanup.